File: tests_ext/src/main/java/com/github/klee0kai/test_ext/inject/di/swcache/SwitchCacheExtComponent.java
package com.github.klee0kai.test_ext.inject.di.swcache;

import com.github.klee0kai.stone.annotations.component.*;
import com.github.klee0kai.test.di.swcache.SwitchCacheComponent;

@Component
public interface SwitchCacheExtComponent extends SwitchCacheComponent {

    @ExtendOf
    void extOf(SwitchCacheComponent parent);

    @GcAllScope
    @SwitchCache(cache = SwitchCache.CacheType.Weak)
    void allWeakExt();

    @GcAllScope
    @SwitchCache(cache = SwitchCache.CacheType.Strong, timeMillis = 100)
    void allStrongFewMillisExt();

    @GcStrongScope
    @SwitchCache(cache = SwitchCache.CacheType.Weak)
    void strongToWeakExt();

}

File: tests_ext/src/main/java/com/github/klee0kai/test_ext/inject/di/swcache/SwitchCacheExtComponent.kt
package com.github.klee0kai.test_ext.inject.di.swcache

import com.github.klee0kai.stone.annotations.component.*
import com.github.klee0kai.test.di.swcache.SwitchCacheComponent

@Component
interface SwitchCacheExtComponent : SwitchCacheComponent {

    @ExtendOf
    fun extOf(parent: SwitchCacheComponent?)

    @GcAllScope
    @SwitchCache(cache = SwitchCache.CacheType.Weak)
    fun allWeakExt()

    @GcAllScope
    @SwitchCache(cache = SwitchCache.CacheType.Strong, timeMillis = 100)
    fun allStrongFewMillisExt()

    @GcStrongScope
    @SwitchCache(cache = SwitchCache.CacheType.Weak)
    fun strongToWeakExt()

}
